src/core/network.ts

```diff
@@ -46,10 +52,20 @@ export class NetworkClient {
    * Logs a user's policy consents to oak-consent-api
    */
   async logConsents(logs: ConsentLog[]) {
-    await fetch(this.config.consentLogUrl, {
+    const url = this.config.consentLogUrl;
+    const body = logs;
+    const response = await fetch(url, {
       method: "POST",
-      body: JSON.stringify(logs),
+      body: JSON.stringify(body),
     });
+
+    if (!response.ok) {
+      throw new OakConsentClientError("Failed to log consents", {
+        url,
+        body,
+        response,
+      });
+    }
```

Comment:
🤔 something for the future, but this got me thinking, perhaps we could be setting a correlation id header that would tie this back to any errors in the consent api

Reply:
Yep, that's a good idea!